File: examples/ionic-vue-qr-code-scanner/android/app/src/main/java/com/dynamsoft/ionic/qrcodescanner/ScanResultContract.java
package com.dynamsoft.ionic.qrcodescanner;

import android.content.Intent;

/**
 * Contract shared between the scanner Activity and the Capacitor plugin.
 */
final class ScanResultContract {

    private ScanResultContract() {
    }

    /** JSON payload produced by a successful scan. */
    static final String EXTRA_RESULTS = "com.dynamsoft.ionic.qrcodescanner.RESULTS";
    /** Error message produced by a failed scan. */
    static final String EXTRA_ERROR = "com.dynamsoft.ionic.qrcodescanner.ERROR";

    static Intent ok(String resultsJson) {
        Intent intent = new Intent();
        intent.putExtra(EXTRA_RESULTS, resultsJson);
        return intent;
    }

    static Intent error(String message) {
        Intent intent = new Intent();
        intent.putExtra(EXTRA_ERROR, message);
        return intent;
    }
}

File: examples/expo-barcode-scanner/modules/expo-dynamsoft-barcode-scanner/android/src/main/java/expo/modules/dynamsoftbarcodescanner/ScanResultContract.kt
package expo.modules.dynamsoftbarcodescanner

import android.content.Intent

/** Contract shared between the scanner Activity and the Expo module. */
object ScanResultContract {

  /** JSON payload produced by a successful scan. */
  const val EXTRA_RESULTS = "expo.modules.dynamsoftbarcodescanner.RESULTS"

  /** Error message produced by a failed scan. */
  const val EXTRA_ERROR = "expo.modules.dynamsoftbarcodescanner.ERROR"

  fun ok(resultsJson: String): Intent =
    Intent().putExtra(EXTRA_RESULTS, resultsJson)

  fun error(message: String): Intent =
    Intent().putExtra(EXTRA_ERROR, message)
}